lib/src/Base/Stat/HMatrixImplementation.cxx

```diff
@@ -178,6 +178,14 @@ HMatrixClusterTree::~HMatrixClusterTree()
 #endif
 }
 
+struct CDeleter
+{
+  void operator ()(void * p)
+  {
+    free(p);
+  }
+};
+
```

Comment:
A kind of HMAT black magic?

Reply:
regular c++ magic ^^

Reply:
My mother told me I would never master C++. She was right (as any mother is)

Reply:
CDeleter helps std::shared_ptr to free the C pointers given by hmat instead of using c++'s delete